### ISSUE TITLE
Freewin plugin - TypeError: setHtml(self, text: Optional[str]): too many arguments

### ISSUE THREAD

**Issue by LewisNeal** [leo-editor/leo-editor#3895]
Leo 6.7.9-devel, devel branch, build 15f6b5fa8d
Python 3.12.2, PyQt version 6.7.0
Windows 10 AMD64 (build 10.0.19045) SP0

Running z-open-freewin command then pressing the [Rendered<-->Plain] toggle gives error:

_Uncaught exception in Leo...
Traceback (most recent call last):
  File "N:\git\leo-editor\leo\plugins\freewin.py", line 967, in switch_and_render
    self.set_and_render(True)
  File "N:\git\leo-editor\leo\plugins\freewin.py", line 960, in set_and_render
    self.browser.setHtml(html, self.url_base)
TypeError: setHtml(self, text: Optional[str]): too many arguments_

**dummy** then displays in the Freewin rendered window.

**Comment by tbpassin** [comment]
Hmmm, I just opened Freewin for that branch and didn't get an error.  I'm using 
PyQt version 6.6.2.  You are using 6.7.0. The docs for 6.7.1 say that the url_base parameter is an allowed optional argument for that method.

I wonder if PyQt 6.7.0 has a bug here.  I'll look into it.

**Comment by tbpassin** [comment]
If you keep getting the error, there is an alternative way I can set the base url, so I could change the code to use it.  It would still be extremely strange, though, to get that error.

**Comment by LewisNeal** [comment]
py -m pip list |find /i "pyqt6"
gives:
PyQt6                         6.7.0
PyQt6-Qt6                     6.7.0
PyQt6-sip                     13.6.0
PyQt6-WebEngine               6.7.0
PyQt6-WebEngine-Qt6           6.7.0

I then reinstalled pyqt6 and pyqt6_webengine with no improvement. Also reinstalled PyQt6-sip.
Same result with:
PyQt6                         6.6.0
PyQt6-Qt6                     6.6.0
PyQt6-sip                     13.6.0
PyQt6-WebEngine               6.6.0
PyQt6-WebEngine-Qt6           6.6.0

and same result with:
PyQt6                         6.6.1
PyQt6-Qt6                     6.6.1
PyQt6-sip                     13.6.0
PyQt6-WebEngine               6.6.0
PyQt6-WebEngine-Qt6           6.6.0

I even downgraded docutils 0.21.2 to 0.21.1 but no change to error.

**Comment by tbpassin** [comment]
I flat out don't understand how this can be.  This changeset is also working on Linux.  But I don't want to fight it.  I'll change to the alternate way the base url is set.  I'll put in a PR for that tomorrow, once I make sure it really works.

Did you try several nodes (I imagine that you did)?  Just in case there is something unusual about one of them - not that it should matter.

**Comment by LewisNeal** [comment]
Yes I did try several nodes. I then tried rendering a node with no body text :)

_Uncaught exception in Leo...
Traceback (most recent call last):
  File "N:\git\leo-editor\leo\plugins\freewin.py", line 967, in switch_and_render
    self.set_and_render(True)
  File "N:\git\leo-editor\leo\plugins\freewin.py", line 955, in set_and_render
    if text[0] == '<':
       ~~~~^^^
IndexError: string index out of range_

I will try downgrading the PyQt6 set to 6.6.2

**Comment by tbpassin** [comment]
I just noticed that myself, just an hour ago, and I'll fix that up too.

BTW, the failed display you saw with the word "dummy" is a simple-minded way to preload docutils and get it up to speed.  The idea was to reduce the latency the first time rendering was asked for.  I'm not sure it really makes any practical difference.  It's one of those things it's probably not worth spending time on, not even to take it out.

**Comment by LewisNeal** [comment]
@tbpassin it appears builds after 45e8ac5 caused the change.
I rolled PyQt6 set back to:
PyQt6                     6.5.3
PyQt6-Qt6                 6.5.3
PyQt6-sip                 13.6.0
PyQt6-WebEngine           6.5.0
PyQt6-WebEngine-Qt6       6.5.0

Then rolled Leo back:
**git checkout 45e8ac5**
Freewin now renders correctly.

updated PyQt6 and PyQt6-Qt6 to 6.7.0
Freewin still renders correctly.

updated PyQt6-WebEngine and PyQt6-WebEngine-Qt6 to 6.7.0
Freewin still renders correctly.

The rendering issue has occurred on 2 Windows machines so I don't feel it is an install glitch. Hope this helps.

**Comment by LewisNeal** [comment]
After git checkout f78fa95 on my windows laptop I can confirm that Freewin does not render.

**Comment by tbpassin** [comment]
Thank you.  That will be very helpful.

**Comment by tbpassin** [comment]
Fixed by https://github.com/leo-editor/leo-editor/pull/3897 (when it is merged)

**Comment by LewisNeal** [comment]
Thankyou @tbpassin it's fixed by #3897
